zerver/lib/events.py

```diff
@@ -294,31 +309,48 @@ def fetch_initial_state_data(
         for property_name in Realm.property_types:
             state["realm_" + property_name] = getattr(realm, property_name)
 
+        if realm_groups_data is None:
+            realm_groups_data = user_groups_in_realm_serialized(
```

Comment:
We do not required complete data for the named user groups, but even if we just want data for memberships of anonymous groups, we would still atleast need IDs of named user groups to know which setting is set to anonymous group and which setting is set to named group.

Reply:
Probably worth adding a comment here to record the understanding. Something like:

> Optimizing opportunity: This fetches more data than we strictly need; we need the membership of the anonymous groups in `realm_setting_group_ids` and the IDs of the NamedUserGroup objects used there, but don't need the other `NamedUserGroup` fields.